src/main/groovy/com/pszymczyk/consul/ConsulStarter.groovy

```diff
@@ -29,9 +29,10 @@ class ConsulStarter {
     private AntUnzip unzip
 
     @PackageScope
-    ConsulStarter(Path dataDir, Path downloadDir, String consulVersion, File portsConfigFile, LogLevel logLevel, ConsulPorts ports) {
+    ConsulStarter(Path dataDir, Path downloadDir, Path configDir, String consulVersion, String customConfig, LogLevel logLevel, ConsulPorts ports) {
```

Comment:
Unfortunately here we should merge `ConsulPorts` with corresponding section of `customConfig` (if exists). We need introduce `ConsulPorts::withXXXPort()` methods for this. I would remove `ConsulPortsBuilder` and move `withXXXPort()` methods to parent class